macros/sql/get_table_types_sql.sql

```diff
@@ -20,3 +20,12 @@
                 else lower(table_type)
             end as "table_type"
 {% endmacro %}
+
+{% macro bigquery__get_table_types_sql() %}
+            case table_type
+                when 'BASE TABLE' then 'table'
+                when 'EXTERNAL TABLE' then 'external'
+                when 'MATERIALIZED VIEW' then 'materializedview'
+                else lower(table_type)
+            end as `table_type`
```

Comment:
You can use `adapter.quote()` here (source: core/default](https://github.com/dbt-labs/dbt-core/blob/899b0ef2241d97d97df5270c20b67be02ec7ca61/core/dbt/adapters/sql/impl.py#L202-L203), [dbt-bigquery](https://github.com/dbt-labs/dbt-bigquery/blob/edebc4728fe6fc6cac79d853387060142e7b9e06/dbt/adapters/bigquery/impl.py#L311-L313))